packages/manager/modules/pci/src/projects/project/storages/databases/database/current-queries/current-queries.controller.js

```diff
@@ -114,6 +114,38 @@ export default class {
     }
   }
 
+  trackAndTerminateQuery(query) {
+    this.trackDashboard(TRACKING_INFO.TERMINATE_QUERY, 'action');
+    return this.DatabaseService.postTerminateQuery(
+      this.projectId,
+      this.database.id,
+      this.database.engine,
+      query.pid,
+    )
+      .then(() =>
+        this.CucCloudMessage.success(
+          this.$translate.instant(
+            'pci_databases_current_queries_datagrid_terminate_action_success',
+            {
+              pid: query.pid,
+            },
+          ),
+          this.messageContainer,
+        ),
+      )
```

Comment:
Shouldn't we refresh the queries once we terminate one ? I'm afraid we won't update the status of the query otherwise.

Reply:
Fixed :)